system/Commands/Translation/LocalizationFinder.php

```diff
@@ -0,0 +1,330 @@
+        }
+
+        if (is_string($optionDir)) {
+            $tempCurrentDir = realpath($currentDir . $optionDir);
+
+            if (false === $tempCurrentDir) {
+                CLI::error('Error: Directory must be located in "' . $currentDir . '"');
+
+                return EXIT_USER_INPUT;
+            }
+
+            if ($this->isSubDirectory($tempCurrentDir, $this->languagePath)) {
+                CLI::error('Error: Directory "' . $this->languagePath . '" restricted to scan.');
+
+                return EXIT_USER_INPUT;
+            }
+
+            $currentDir = $tempCurrentDir;
+        }
+
+        $this->process($currentDir, $currentLocale);
+
+        CLI::write('All operations done!');
+
+        return EXIT_SUCCESS;
+    }
+
+    /**
+     * @param SplFileInfo|string $file
+     */
+    private function findTranslationsInFile($file): array
+    {
+        $foundLanguageKeys = [];
+
+        if (is_string($file) && is_file($file)) {
+            $file = new SplFileInfo($file);
+        }
+
+        $fileContent = file_get_contents($file->getRealPath());
+        preg_match_all('/lang\(\'([._a-z0-9\-]+)\'\)/ui', $fileContent, $matches);
+
+        if ([] === $matches[1]) {
+            return [];
+        }
+
+        foreach ($matches[1] as $phraseKey) {
+            $phraseKeys = explode('.', $phraseKey);
+
+            /**
+             * Language key not have Filename or Lang key
+             */
+            if (count($phraseKeys) < 2) {
+                continue;
+            }
+
+            $languageFileName   = array_shift($phraseKeys);
+            $isEmptyNestedArray = (! empty($languageFileName) && empty($phraseKeys[0]))
+                || (empty($languageFileName) && ! empty($phraseKeys[0]))
+                || (empty($languageFileName) && empty($phraseKeys[0]));
+
+            if ($isEmptyNestedArray) {
+                continue;
+            }
+
+            /**
+             * Language key as string
+             */
+            if (count($phraseKeys) === 1) {
+                /**
+                 * Add found language keys to temporary array
+                 */
+                $foundLanguageKeys[$languageFileName][$phraseKeys[0]] = $phraseKey;
+            } else {
+                $childKeys                            = $this->buildMultiArray($phraseKeys, $phraseKey);
+                $foundLanguageKeys[$languageFileName] = array_replace_recursive($foundLanguageKeys[$languageFileName] ?? [], $childKeys);
+            }
+        }
+
+        return $foundLanguageKeys;
+    }
+
+    private function isIgnoredFile(SplFileInfo $file): bool
+    {
+        if ($file->isDir() || $this->isSubDirectory($file->getRealPath(), $this->languagePath)) {
+            return true;
+        }
+
+        return 'php' !== $file->getExtension();
+    }
+
+    private function templateFile(array $language = []): string
+    {
+        if ([] !== $language) {
+            $languageArrayString = $this->languageKeysDump($language);
```

Comment:
I don;t think you need the extra private method here. You can just put the var export here.

Reply:
This is needed right now. Because we haven't decided on the formatting. If you need to execute preg_replace() or something else, this is where it belongs.

Reply:
Ok, deleted